app/models/service_plan.rb

```diff
@@ -10,4 +11,45 @@ class ServicePlan < ApplicationRecord
   has_many   :service_instances
 
   acts_as_taggable
+
+  def order(catalog_id, additional_parameters)
+    payload = external_payload(catalog_id, additional_parameters)
+
+    response = service_catalog_client.order_service_plan(payload)
+
+    parsed_data = JSON.parse(response.body)
+    task = Task.create!(:tenant => tenant, :context => parsed_data['metadata']['selfLink'])
```

Comment:
How about:
```
:context => {:service_instance => {:source_id => ext_management_system.id, :source_ref => parsed_data['metadata']['selfLink']}
```

This will provide the caller enough information to find the new object.